### PR TITLE
chore(deployment): pin ubuntu version used by github (#30871)

### LINKED ISSUE #30871: pin ubuntu version used by github
github is changing version of ubuntu they're going to use in github runners.  They're retagging `latest` tag.

We want to get in front of this and pin so we have reproducible builds and control the cutover to new versions.

## Acceptance Criteria 
1. where we use `latest` pin to the current version we're using 
2. bring forward the ones we have pinned to older versions that are being deprecated.  They should go to the same version as the rest from AC `1`

### PR DESCRIPTION
**Pull request by dcolina**
### Proposed Changes

* This pull request proposes updating all Ubuntu runners in our CICD pipelines to version 24.04.
* A new GitHub repository variable is needed. `UBUNTU_RUNNER_VERSION` must be set to `24.04`; otherwise, workflows will default to using `22.04`. 

## Details
We conducted thorough testing with the following setups:

*Ubuntu 22.04:*

- Executed main CICD workflows (PR, Merge Queue, Trunk, and Nightly) on fork.
All runs completed successfully without any unusual behavior.

*Ubuntu 24.04:*

- Repeated the same tests with runners updated to 24.04.
Similarly, all workflows ran correctly, with no anomalies observed.

### Proposal

- Given the successful results in both cases, we propose setting the runners directly to version 24.04 to streamline the update process.

### Next Steps

- Merge this PR to update the Ubuntu runners.
- Monitor post-deployment to ensure stability across all workflows.

### Fixes
#30871 

This PR fixes: #30871

### CODE REVIEW

**Review by yolabingo — APPROVED** [review]

**Comment by victoralfaro-dotcms** [comment]
@dcolina thank you for all these changes.
This has grown enough to make me wonder if we could take this opportunity to actually store this ubuntu version as a repo `ENVIRONMENT VARIABLE`.
At the workflows we could reference it like this:
```
  runs-on: ubuntu-{{ vars.UBUNTU_MAIN_VERSION }} # or any other name since I'm terrible at naming
```
This way we could make the change in one single place without introducing any code changes in the future. We can also manage several versions since some workflows (can't recall which ones) work better with specific ubuntu versions.

**Review by nollymar — APPROVED** [review]

File: .github/actions/core-cicd/api-limits-check/Readme.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ on:
 
 jobs:
   check-rate-limits:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/api-limits-check/Readme.md:27]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/cleanup-runner/Readme.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ Here's an example of how to incorporate this action into your workflow:
 ```yaml
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/cleanup-runner/Readme.md:43]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/deployment/deploy-javadoc/README.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ on:
 
 jobs:
   deploy-javadoc:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/deployment/deploy-javadoc/README.md:56]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/deployment/deploy-javascript-sdk/README.md [hunk 4]
```diff
@@ -71,7 +71,7 @@ on:
 
 jobs:
   publish-sdk:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/deployment/deploy-javascript-sdk/README.md:74]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/deployment/deploy-jfrog/README.md [hunk 5]
```diff
@@ -32,7 +32,7 @@ on:
 
 jobs:
   deploy:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/deployment/deploy-jfrog/README.md:35]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/maven-job/README.md [hunk 6]
```diff
@@ -71,7 +71,7 @@ Here's an example of how to incorporate this action into your workflow:
 ```yaml
 jobs:
   build-and-test:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/maven-job/README.md:74]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/notification/notify-slack/README.md [hunk 7]
```diff
@@ -24,7 +24,7 @@ on:
 
 jobs:
   notify:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/notification/notify-slack/README.md:27]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/prepare-runner/README.md [hunk 8]
```diff
@@ -53,7 +53,7 @@ Here's an example of how to incorporate this action into your workflow:
 ```yaml
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/prepare-runner/README.md:56]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/actions/core-cicd/setup-java/README.md [hunk 9]
```diff
@@ -52,7 +52,7 @@ Here's an example of how to incorporate this action into your workflow:
 ```yaml
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/actions/core-cicd/setup-java/README.md:55]: Suggest updating this to reflect how you've implemented with fallback, or even with no fallback at all

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: .github/workflows/legacy-release_sbom-generator.yaml [hunk 10]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
   scan:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline .github/workflows/legacy-release_sbom-generator.yaml:14]: Why not use the variable here with fallback?

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: tools/dotcms-cli/action/.github/workflows/main.yml [hunk 11]
```diff
@@ -5,7 +5,7 @@ on: [push]
 jobs:
 
   dotcli-runner:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline tools/dotcms-cli/action/.github/workflows/main.yml:8]: Why not use variable with fallback?

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

File: tools/dotcms-cli/action/.github/workflows/main.yml [hunk 12]
```diff
@@ -30,7 +30,7 @@ jobs:
 
   sync-with-dotcms:
     needs: dotcli-runner
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (sfreudenthaler) [inline tools/dotcms-cli/action/.github/workflows/main.yml:33]: Why not use variable with fallback?

```suggestion
    runs-on: ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}
```

**Review by sfreudenthaler — CHANGES_REQUESTED** [review]
## Requests
- Requested some changes with the readme's 
- Did you manually test the fallback in the forked repo to make sure it works?

## Questions that are not blockers
- When will we remove the fallback, or at least update it to fallback to 24.04 if no var provided?
- is there some way to define the full runs on value `ubuntu-${{ vars.UBUNTU_RUNNER_VERSION || '22.04' }}` once and have it inherited downstream?  Just trying to keep pushing to fewer places we have to repeat ourselves :) 
```

File: .github/workflows/legacy-release_sbom-generator.yaml (hunk 10 above)
Comment (dcolina) [reply .github/workflows/legacy-release_sbom-generator.yaml:14]: I think this workflow should be removed. It's deprecated. The SBOM generation is now an Action, and it's called from the release process.

File: tools/dotcms-cli/action/.github/workflows/main.yml (hunk 11 above)
Comment (dcolina) [reply tools/dotcms-cli/action/.github/workflows/main.yml:8]: It isn't part of our CICD process. It's an example of using dotCLI along with Github Actions.

File: tools/dotcms-cli/action/.github/workflows/main.yml (hunk 12 above)
Comment (dcolina) [reply tools/dotcms-cli/action/.github/workflows/main.yml:33]: It isn't part of our CICD process. It's an example of using dotCLI along with Github Actions.

**Comment by dcolina** [comment]
- All the readme file (examples) have been updated.
- Fallback was manually tested in the forked repo.
- Fallback has been set to `24.04`
- There is no straightforward or simple way to define the `runs-on` configuration in a single location and reference it across all workflows in the project. The available alternatives would require significantly more changes than we’d prefer, and they do not fully provide the desired solution—at least from a development perspective.

**Review by sfreudenthaler — APPROVED** [review]
Looks great.  Thanks for the changes and thoughts

**Pull request merged**